src/isomorphic/classic/element/ReactElementType.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ * @providesModule ReactElementType
+ */
+
+'use strict';
+
+import type { ReactInstance } from 'ReactInstanceType';
+
+export type Source = {
+  fileName: string,
+  lineNumber: number,
+};
+
+export type ReactElement = {
```

Comment:
I'm not sure how to name this one since it's going to conflict with the `ReactElement` name. I've thought about appending `Type` but it kind of sucks to have all the types have Type at the end... In a normal codebase, the same name works because flow assumes instanceof. But in the React codebase, a lot of things return random objects with weird constructions :p